src/artemis/external_interaction/ispyb/store_in_ispyb.py

```diff
@@ -233,77 +244,198 @@ def _store_data_collection_table(self, data_collection_group_id: int) -> int:
             "file_template"
         ] = f"{self.detector_params.prefix}_{self.run_number}_master.h5"
 
-        return self.mx_acquisition.upsert_data_collection(list(params.values()))
+        return mx_acquisition.upsert_data_collection(list(params.values()))
 
-    def _store_position_table(self, dc_id: int) -> int:
-        params = self.mx_acquisition.get_dc_position_params()
 
-        params["id"] = dc_id
+class StoreRotationInIspyb(StoreInIspyb):
+    ispyb_params: RotationIspybParams
+    data_collection_id: int | None = None
+
+    def __init__(self, ispyb_config, parameters: RotationInternalParameters) -> None:
+        self.full_params: RotationInternalParameters = parameters
+        self.ispyb_params: RotationInternalParameters = (
+            parameters.artemis_params.ispyb_params
+        )
+        self.detector_params = parameters.artemis_params.detector_params
+        self.omega_start = self.detector_params.omega_start
+
+        if self.ispyb_params.xtal_snapshots_omega_start:
+            self.xtal_snapshots = self.ispyb_params.xtal_snapshots_omega_start
+        else:
+            self.xtal_snapshots = []
+            LOGGER.warning("No xtal snapshot paths sent to ISPyB!")
+        super().__init__(ispyb_config, "SAD")
+
+    def _mutate_data_collection_params_for_experiment(
```

Comment:
Should: I think a few additional simple tests just covering this function would be good